### PR TITLE
Add table docs

### PR DESCRIPTION
**Pull request by sukhrajghuman**
https://designsystem.apps.y.cld.gov.au/

### CODE REVIEW

File: content/components/table/code/react-props.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+| Prop name           | Type        | Description |
+| ------------------- | ----------- | ----------- |
+| `caption`           | string      | The table caption summary |
+| `headers`           | array      | The table column headers  |
+| `headers[0].title`  | string     | The table column header title text |
+| `headers[0].type`   | string     | Type of the header, can be either `text` or `numeric` for left or right alignment respectively. |
+| `headers[0].width`  | string     | The width of the header. Can be either `10`, `20`, `25`, `33`, `50` or `75` |
+| `headers[0].key`    | string     | The key used in the corresponding row of data|
+| `headers[0].render` | function   | A function used to customly render the corresponding data cells|
```
Comment (gordongrace) [inline content/components/table/code/react-props.md]: customly?

File: content/components/table/code/react-props.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+All other props are spread into the component
+
+
+### AUtableHeader
+
+| Prop name   | Type        | Description |
+| ----------- | ----------- | ----------- |
+| `title`        |  string    | The table column header title text |
+| `type`        |  string    | Type of the header, can be either `text` or `numeric` for left or right alignment respectively. |
```
Comment (gordongrace) [inline content/components/table/code/react-props.md]: Would we want to support a header type that was right-aligned text?

File: content/components/table/code/react-props.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+| `width`        |  string    | The width of the header. Can be either `10`, `20`, `25`, `33`, `50` or `75` |
```
Comment (gordongrace) [inline content/components/table/code/react-props.md]: %ge?

File: content/components/table/code/react-props.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+| Prop name   | Type        | Description |
+| ----------- | ----------- | ----------- |
+| `className`        | string     | An additional class, optional |
+
+
+### AUtableCaption
+
+| Prop name   | Type        | Description |
+| ----------- | ----------- | ----------- |
+| `tableCaption`        | string     | Title of the caption |
```
Comment (gordongrace) [inline content/components/table/code/react-props.md]: Suggest: "Caption (or title) of the table"

File: content/components/table/default.md [hunk 5]
```diff
@@ -1,3 +1,58 @@
...
+            <AUtable
+                caption="Population of Australian states and territories, December 2015"
+                headers={simpleHeaders}
+                data={simpleData}
+            />
+        </AUtableResponsiveWrapper>
+---
+## Default table
+
+For data tables with fewer than four rows, use a table without stripes. Align text columns and corresponding data cells to the left. When comparing numbers in a column, align data cells and column headers to the right.
```
Comment (gordongrace) [inline content/components/table/default.md]: Do we want to provide a rationale for this recommendation?

File: content/components/table/discussion/community-intro.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+layout: component/discussion-callout
+btntext: Community
+btnurl: https://community.digital.gov.au/t/
+btntype: secondary
+---
+
+Our components are a collaborative effort, we welcome constructive conversation and contribution.
```
Comment (gordongrace) [inline content/components/table/discussion/community-intro.md]: ...collaborative effort. We welcome...

File: content/components/table/default.md [hunk 7]
```diff
@@ -1,3 +1,58 @@
...
-[community forum](https://community.digital.gov.au/t/table).
+---
+layout: component/code-demo
+iframe: examples/example-default
+iframeFullwidth: true
+code:
+  - HTML: |
+        <table class="au-table">
+            <caption class="au-table__caption">Population of Australian states and territories, December 2015</caption>
+            <thead class="au-table__head">
```
Comment (gordongrace) [inline content/components/table/default.md]: Anticipating questions on <col> and <colgroup> - do we want to explain why these weren't in scope of the initial release?

File: content/components/table/default.md [hunk 8]
```diff
@@ -1,3 +1,58 @@
...
+                    <td class="au-table__cell au-table__cell--numeric">5,996,400</td>
+                    <td class="au-table__cell au-table__cell--numeric">2.5%</td>
+                </tr>
+                <tr class="au-table__row">
+                    <td class="au-table__cell">Tasmania</td>
+                    <td class="au-table__cell au-table__cell--numeric">517,400</td>
+                    <td class="au-table__cell au-table__cell--numeric">4%</td>
+                </tr>
+            </tbody>
+        </table>
```
Comment (gordongrace) [inline content/components/table/default.md]: Anticipating questions re: `<tfoot>`. See comment above re:col and colgroup.

File: content/components/table/width.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                caption="Population of Australian states and territories, December 2015"
+                headers={simpleHeaders}
+                data={simpleData}
+            />
+        </AUtableResponsiveWrapper>
+
+---
+## Custom header widths
+
+Use custom widths based on the expected length of the data under each corresponding column. By default the table is full width, however the table can be used with the grid to take up a desired amount of horizontal space on the page. Wrapping the `table` element in a `<div class="au-table__wrapper">` allows for horizontal scrolling on smaller devices, this is especially useful for tables with a larger number of columns.
```
Comment (gordongrace) [inline content/components/table/width.md]: ...on smaller devices. This is especially useful...

File: content/components/table/rationale/alternating-rows.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+---
+layout: component/rationale.js
+heading: Alternating rows
+---
+
+For a data table with large amounts of rows, highlighting alternating rows helps [increase readability and scanning speed](http://alistapart.com/article/zebrastripingmoredataforthecase/). We've chosen for the bottom row to be highlighted using the `nth-last-child(odd)` selector, since this 'visually completes' the table.
```
Comment (gordongrace) [inline content/components/table/rationale/alternating-rows.md]: "We've elected to highlight the bottom row using... to provide the table with visual completeness."

File: content/components/table/code/react-props.md (hunk 1 above)
Comment (sukhrajghuman) [inline content/components/table/code/react-props.md]:
```suggestion
| `headers[0].render` | function   | A function used to customise rendering for the corresponding data cells|
```

File: content/components/table/code/react-props.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+#### AUtable
+
+| Prop name           | Type        | Description |
+| ------------------- | ----------- | ----------- |
+| `caption`           | string      | The table caption summary |
+| `headers`           | array      | The table column headers  |
+| `headers[0].title`  | string     | The table column header title text |
+| `headers[0].type`   | string     | Type of the header, can be either `text` or `numeric` for left or right alignment respectively. |
+| `headers[0].width`  | string     | The width of the header. Can be either `10`, `20`, `25`, `33`, `50` or `75` |
```
Comment (sukhrajghuman) [inline content/components/table/code/react-props.md]:
```suggestion
| `headers[0].width`  | string     | The width (%) of the header. Can be either `10`, `20`, `25`, `33`, `50` or `75` |
```

File: content/components/table/code/react-props.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+| Prop name   | Type        | Description |
+| ----------- | ----------- | ----------- |
+| `className`        | string     | An additional class, optional |
+
+
+### AUtableCaption
+
+| Prop name   | Type        | Description |
+| ----------- | ----------- | ----------- |
+| `tableCaption`        | string     | Title of the caption |
```
Comment (sukhrajghuman) [inline content/components/table/code/react-props.md]:
```suggestion
| `tableCaption`        | string     | Caption (or title) of the table |
```

**Pull request merged**